src/com/twitter/intellij/pants/macro/FilePathRelativeToBuiltRootMacro.java

```diff
@@ -6,37 +6,80 @@
 import com.intellij.ide.macro.Macro;
 import com.intellij.openapi.actionSystem.CommonDataKeys;
 import com.intellij.openapi.actionSystem.DataContext;
+import com.intellij.openapi.project.Project;
 import com.intellij.openapi.util.io.FileUtil;
 import com.intellij.openapi.vfs.VfsUtil;
 import com.intellij.openapi.vfs.VirtualFile;
+import com.twitter.intellij.pants.bsp.JarMappings;
+import com.twitter.intellij.pants.bsp.PantsBspData;
+import com.twitter.intellij.pants.bsp.PantsTargetAddress;
 import com.twitter.intellij.pants.util.PantsUtil;
+import org.jetbrains.annotations.NotNull;
 
+import java.nio.file.Path;
 import java.util.Optional;
+import java.util.Set;
 
 public class FilePathRelativeToBuiltRootMacro extends Macro {
   /**
    * @return corresponding name of this macro
    */
+  @NotNull
   @Override
   public String getName() {
     return "PantsFilePathRelativeToBuildRoot";
   }
 
+  @NotNull
   @Override
   public String getDescription() {
     return "Relative path from build root";
   }
 
+  private Optional<String> jarFilePath(Project project, VirtualFile vFile) {
+    Optional<VirtualFile> jarFile = JarMappings.getParentJar(vFile);
+    if(jarFile.isPresent()) {
+      JarMappings mappings = JarMappings.getInstance(project);
+      Optional<Path> target = mappings
```

Comment:
Do we guess the root directory from the name of the jar file? I think it can be more robust if fastpass writes the root directory in the jar file, around here 

https://github.com/scalameta/fastpass/blob/df9359fa7756c6e7f0f062af56fe617201897876/fastpass/src/main/scala/scala/meta/internal/fastpass/pantsbuild/SourcesJarBuilder.scala#L24

If that info doesn't exist in the jar file then we can fallback to this heuristic.

Reply:
Ok, no it tries to get this info from `*jar!/META-INF/fastpass/source-root`